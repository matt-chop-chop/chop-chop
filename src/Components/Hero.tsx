import React from "react";
import { Flex, Heading, Image, Text } from "@chakra-ui/react";

export type HeroProps = {
  title: string;
  message: string;
};

const Hero = ({ title, message }: HeroProps) => {
  return (
    <Flex alignItems="center" flexDirection="column" textAlign="center">
      <Image alt="Chop Chop logo" height="128px" src="/images/logo.png" />
      <Heading
        variant={["heading-medium", "heading-large", "heading-large"]}
        maxWidth={["none", "880px", "880px"]}
        mt={[3, 5, 5]}
      >
        {title}
      </Heading>
      <Text
        variant={["body", "body-large", "body-large"]}
        mt={[3, 5, 5]}
        maxWidth={["none", "740px", "740px"]}
        px={3}
      >
        {message}
      </Text>
    </Flex>
  );
};

export default Hero;
