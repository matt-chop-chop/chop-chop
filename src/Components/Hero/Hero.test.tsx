import { shallow } from "enzyme";
import React from "react";
import Hero, { HeroProps } from "./Hero";

const props: HeroProps = {
  title: "Hero Test Title",
  message: "This is a test",
};

describe("Hero", () => {
  it("is exported from Hero", () => {
    expect(Hero).toBeDefined();
  });

  it("renders as expected ", () => {
    const wrapper = shallow(<Hero {...props} />);

    expect(wrapper).toMatchSnapshot();
  });
});
