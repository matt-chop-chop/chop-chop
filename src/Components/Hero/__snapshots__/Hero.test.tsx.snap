// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hero renders as expected  1`] = `
<Flex
  alignItems="center"
  flexDirection="column"
  textAlign="center"
>
  <Image
    alt="Chop Chop logo"
    height="128px"
    src="/images/logo.png"
  />
  <Heading
    as="h1"
    maxWidth={
      [
        "none",
        "880px",
        "880px",
      ]
    }
    mt={
      [
        3,
        5,
        5,
      ]
    }
    variant={
      [
        "heading-medium",
        "heading-large",
        "heading-large",
      ]
    }
  >
    Hero Test Title
  </Heading>
  <Text
    maxWidth={
      [
        "none",
        "740px",
        "740px",
      ]
    }
    mt={
      [
        3,
        5,
        5,
      ]
    }
    px={3}
    variant={
      [
        "body",
        "body-large",
        "body-large",
      ]
    }
  >
    This is a test
  </Text>
</Flex>
`;
